docs: update PHPUnit testing README.md with corrections #62065

q0rban opened the pull request:
<!--
Hi there! Thanks for contributing to WordPress!

Pull Requests in this GitHub repository **must** be linked to a ticket in the WordPress Core Trac instance (https://core.trac.wordpress.org), and are only used for code review. **No pull requests will be merged on GitHub.**

See the WordPress Handbook page on using PRs for Code Review more information: https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/

If this is your first time contributing, you may also find reviewing these guides first to be helpful:
- FAQs for New Contributors: https://make.wordpress.org/core/handbook/tutorials/faq-for-new-contributors/
- Contributing with Code Guide: https://make.wordpress.org/core/handbook/contribute/
- WordPress Coding Standards: https://make.wordpress.org/core/handbook/best-practices/coding-standards/
- Inline Documentation Standards: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/
- Browser Support Policies: https://make.wordpress.org/core/handbook/best-practices/browser-support/
- Proper spelling and grammar related best practices: https://make.wordpress.org/core/handbook/best-practices/spelling/
-->

Correct the in-repo PHPUnit testing instructions, and switch to Markdown.

Trac ticket: https://core.trac.wordpress.org/ticket/62065

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

Commit:
docs: update PHPUnit testing README.md with corrections #62065

ironprogrammer commented on tests/phpunit/README.md:
```diff
@@ -0,0 +1,33 @@
+
+## Setup
+
+1. Check out the wordpress-develop repository
+
+       git clone git://develop.git.wordpress.org/ wordpress-develop
```
```suggestion
1. Check out the [`wordpress-develop` repository](https://github.com/WordPress/wordpress-develop):

       git clone git@github.com:WordPress/wordpress-develop.git
```
I wonder if this should instead refer to the Github repo, as PRs there are the typical way to contribute patches and tests.

ironprogrammer commented on tests/phpunit/README.md:
```diff
@@ -0,0 +1,33 @@
+       git clone git://develop.git.wordpress.org/ wordpress-develop
+       cd wordpress-develop
+
+2. Run `composer update --with-all-dependencies`
```
```suggestion
2. Run `composer update --with-all-dependencies`.
```
Nit.

ironprogrammer commented on tests/phpunit/README.md:
```diff
@@ -0,0 +1,33 @@
+3. Create a clean MySQL database and user. **DO NOT USE AN EXISTING DATABASE** or you will lose data, guaranteed.
+4. Copy `wp-tests-config-sample.php` to `wp-tests-config.php`, edit it to include your testing database credentials.
+5. Run the tests from the repository root:
+   - To execute a particular test:
+
+         $ vendor/bin/phpunit tests/phpunit/tests/test_case.php
+   - To execute all tests:
+
+         $ vendor/bin/phpunit
```
```suggestion
- To execute a particular test:

vendor/bin/phpunit tests/phpunit/tests/test_case.php

- To execute all tests:

vendor/bin/phpunit

```
Suggest dropping the terminal prompt character. For these commands, please substitute the block formatting spaces by wrapping with three backticks for [consistency with other docs](https://github.com/WordPress/wordpress-develop/blob/552077a27589294b2cf6097205a4045585016281/README.md?plain=1#L50-L55) (I'm unsure how to add these within a backticked suggestion block 😅).

q0rban replied:
I have switched to the three backticks, however this still requires indentation or else the list formatting gets messed up. I hope that's okay! :)

ironprogrammer commented on tests/phpunit/README.md:
```diff
@@ -0,0 +1,33 @@
+1. Check out the wordpress-develop repository
+
+       git clone git://develop.git.wordpress.org/ wordpress-develop
+       cd wordpress-develop
```
```suggestion
cd wordpress-develop
```
For this and the `git clone` line above it, please wrap with three backticks instead of spaces ([for example](https://github.com/WordPress/wordpress-develop/blob/552077a27589294b2cf6097205a4045585016281/README.md?plain=1#L50-L55)).

ironprogrammer reviewed (changes requested):
Thanks for running through and testing the PHPUnit setup, and helping improve the docs, @q0rban! I've left a few suggestions below.

Also, don't forget to [link your GH and WPORG accounts](https://github.com/WordPress/wordpress-develop/pull/7389#issuecomment-2357211040) so you can get your props!

Commit:
fix: address PR feedback

q0rban commented:
@ironprogrammer thank you for the review! Apologies for the delay in my addressing your feedback. I believe I've got your suggestions addressed, however I did run into a slight issue switching to backticks while preserving proper indentation. Let me know if you would like me to try to format this in a different way.

ironprogrammer reviewed (approved):
The updates look good, @q0rban!

(One thing I noticed this round was that on line 5 the link for "PHP: PHP Unit" has a space there, but it should be "PHP: PHPUnit". I don't want to block this any further, so just mentioning it and giving my thumbs up in the meantime! 👍🏻 I can drop a short note on the Trac ticket and add it to the 6.8 milestone once that's updated.)

Commit:
fix: typo in title of handbook page

q0rban commented:
I couldn't resist fixing the typo. Good catch!

desrosj commented on tests/phpunit/README.md:
````diff
@@ -0,0 +1,35 @@
+	git clone git@github.com:WordPress/wordpress-develop.git
+	cd wordpress-develop
+	```
+2. Run `composer update --with-all-dependencies`.
````
```suggestion
2. Run `composer update -W`.
```

This is the shorthand for this option.

q0rban replied:
Is that preferred here? I personally default to [avoiding short options](https://architecture.lullabot.com/adr/20211006-avoid-command-aliases/), especially in documentation. Happy to make this change if this is not the  case for WordPress documentation. ☺️

desrosj commented on tests/phpunit/README.md:
````diff
@@ -0,0 +1,35 @@
+	cd wordpress-develop
+	```
+2. Run `composer update --with-all-dependencies`.
+3. Create a clean MySQL database and user. **DO NOT USE AN EXISTING DATABASE** or you will lose data, guaranteed.
````
```suggestion
3. Create a clean MySQL database and user. **DO NOT USE AN EXISTING DATABASE.** The test database is emptied and repopulated every time tests are run.
```

desrosj commented on tests/phpunit/README.md:
````diff
@@ -0,0 +1,35 @@
+	```
+2. Run `composer update --with-all-dependencies`.
+3. Create a clean MySQL database and user. **DO NOT USE AN EXISTING DATABASE** or you will lose data, guaranteed.
+4. Copy `wp-tests-config-sample.php` to `wp-tests-config.php`, edit it to include your testing database credentials.
````
```suggestion
4. Copy `wp-tests-config-sample.php` to `wp-tests-config.php` and enter your database credentials. **Again, use a separate database.**
```

desrosj reviewed (changes requested):
I left some suggestions, but while reviewing I realized we're creating an additional place for our documentation to become out of date.

What if we included the intro paragraph and then just linked out to the handbook for setup instructions? Is there a reason we must include setup instructions in this file?

q0rban committed:
fix: apply suggestions from code review by @desrosj

Co-authored-by: Jonathan Desrosiers <359867+desrosj@users.noreply.github.com>

q0rban committed:
link off to Handbook pages for all PHPUnit documentation

q0rban commented:
Updated with a simplified version that links away to relevant Handbook pages.

ironprogrammer reviewed (approved):
LGTM 👍🏻

juampick reviewed (approved):
Looks good to me!
